### PR TITLE
DRAFT: Update cluster certificates

### PR DESCRIPTION
**Pull request by masnax**

### CODE REVIEW

File: lxd/api_cluster.go [hunk 1]
```diff
@@ -1615,6 +1623,120 @@ func clusterNodeDelete(d *Daemon, r *http.Request) response.Response {
...
+			}
+		}
+
+		// Get the new cluster certificate struct
+		cert, err := util.LoadClusterCert(d.os.VarDir)
+		if err != nil {
+			return response.BadRequest(fmt.Errorf("oopsies"))
+		}
+
+		defer updateCertificateCacheFromLocal(d, cert)
```
Comment (stgraber) [inline lxd/api_cluster.go:1728]: Shouldn't be needed as that's just caching client certs.

The fact that this function takes the cluster cert feels like a bug, should be removed in a separate commit :)

File: lxd/api_cluster.go [hunk 2]
```diff
@@ -1615,6 +1623,120 @@ func clusterNodeDelete(d *Daemon, r *http.Request) response.Response {
...
+		defer d.endpoints.NetworkUpdatCert(cert)
```
Comment (stgraber) [inline lxd/api_cluster.go:1729]: Should be called synchronously not deferred.

File: lxd/api_cluster.go [hunk 3]
```diff
@@ -1615,6 +1623,120 @@ func clusterNodeDelete(d *Daemon, r *http.Request) response.Response {
...
+		defer d.gateway.Restart(cert)
```
Comment (stgraber) [inline lxd/api_cluster.go:1730]: Should not be needed, we should make NetworkUpdateCert actually fully update the cert.

File: shared/api/cluster.go [hunk 4]
```diff
@@ -207,3 +207,20 @@ type ClusterMemberPut struct {
 	// API extension: clustering_description
 	Description string `json:"description" yaml:"description"`
 }
+
+// ClusterCertificatesPost represents the certificate and key pair for all
+// members in a LXD Cluster
```
Comment (stgraber) [inline shared/api/cluster.go:212]: Should be kept on one line even if a bit long.

File: shared/api/cluster.go [hunk 5]
```diff
@@ -207,3 +207,20 @@ type ClusterMemberPut struct {
...
+//
+// swagger:model
+//
+// API extension: clustering_update_certs
+type ClusterCertificatesPost struct {
+
```
Comment (stgraber) [inline shared/api/cluster.go:218]: Drop blank line

File: lxd/api_cluster.go (hunk 1 above)
Comment (tomponline) [reply lxd/api_cluster.go:1728]: Well originally it was required as the comment says '
If no local trusted server certificates available and is clustered, then loads network certificate into trusted store' but I can't see it being used now so looks like I changed the approach for per member cert upgrades and both the arg and comment are redundant now

Comment (tomponline) [reply lxd/api_cluster.go:1728]: Fixed here https://github.com/lxc/lxd/pull/8882

File: lxd/api_cluster.go (hunk 2 above)
Comment (tomponline) [reply lxd/api_cluster.go:1729]: Also this cant build as the function name is wrong.
